src/tests/profiler/native/nongcheap/nongcheap.cpp

```diff
@@ -209,9 +215,11 @@ HRESULT NonGcHeapProfiler::GarbageCollectionFinished()
             nonGcObjectsEnumerated++;
         }
 
-        if (nonGcObjectsEnumerated != _nonGcHeapObjects)
+        if (nonGcObjectsEnumerated == 0)
```

Comment:
What if we added a p/invoke in the test code [here](https://github.com/dotnet/runtime/blob/cd45c9b8f05dee30e4040e596f29685289dc4e70/src/tests/profiler/gc/nongcheap.cs#L29) informing the native profiler that it is safe to check `nonGcObjectsEnumerated==_nonGcHeapObjects` in the next GC that occurs? I am guessing the check was failing because a GC happened to be running inter-mixed with non-GC allocations.

Reply:
Added a pinvoke. I presume it's still not a 100% guarantee as VM may also produce nongc allocations in non-determinate timings, but presumably the probability of that is low.